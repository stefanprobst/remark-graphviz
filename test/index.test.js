import * as fs from 'fs'
import * as path from 'path'

import toHtml from 'rehype-stringify'
import fromMarkdown from 'remark-parse'
import toHast from 'remark-rehype'
import unified from 'unified'

import graphviz from '../src/index.js'

function createProcessor(options) {
  const processor = unified()
    .use(fromMarkdown)
    .use(graphviz, options)
    .use(toHast)
    .use(toHtml)
  return processor
}

const doc = fs.readFileSync(
  path.join(process.cwd(), 'test', '__fixtures__', 'digraph.md'),
  { encoding: 'utf-8' },
)

describe('remark-graphviz', () => {
  it('should generate figure element with generated svg', async () => {
    const result = String(await createProcessor().process(doc))
    expect(result).toMatchSnapshot()
  })

  describe('outputPath option', () => {
    it('should generate image element and svg file', async () => {
      const options = { outputFolder: './test/__output__', publicFolder: '/' }
      const result = String(await createProcessor(options).process(doc))
      expect(result).toMatchSnapshot()
      expect(
        fs.existsSync(
          path.join(
            process.cwd(),
            options.outputFolder,
            'f7348fe1104f7dab79c8d931cbece7f1.svg',
          ),
        ),
      ).toBe(true)
    })

    it.todo('should show error message when "publicFolder" option is missing')
  })

  describe('optimize option', () => {
    it('should optimize generated svg element with svgo', async () => {
      const options = { optimize: true }
      const result = String(await createProcessor(options).process(doc))
      expect(result).toMatchSnapshot()
    })

    it('should optimize generated svg file with svgo', async () => {
      const options = {
        outputFolder: './test/__output__',
        publicFolder: '/',
        optimize: true,
      }
      const result = String(await createProcessor(options).process(doc))
      expect(result).toMatchSnapshot()
      expect(
        fs.existsSync(
          path.join(
            process.cwd(),
            options.outputFolder,
            '201e5d6a36c0d17cc401665c7988024f.svg',
          ),
        ),
      ).toBe(true)
    })
  })
})
