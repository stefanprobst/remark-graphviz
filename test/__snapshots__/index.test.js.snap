// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-graphviz optimize option should optimize generated svg element with svgo 1`] = `
"<p>Some text before.</p>
<figure><svg viewBox=\\"0 0 224 409.01\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g class=\\"graph\\"><path fill=\\"#fff\\" stroke=\\"transparent\\" d=\\"M0 409.01V0h224v409.01H0z\\"></path><g class=\\"cluster\\"><path fill=\\"#d3d3d3\\" stroke=\\"#d3d3d3\\" d=\\"M12 340.8V48h90v292.8H12z\\"></path><text text-anchor=\\"middle\\" x=\\"53\\" y=\\"-340.41\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">Hello</text></g><g class=\\"cluster\\"><path fill=\\"none\\" stroke=\\"#00f\\" d=\\"M142 340.8V48h70v292.8h-70z\\"></path><text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-340.41\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">World</text></g><g class=\\"node\\"><path fill=\\"#fff\\" stroke=\\"#fff\\" d=\\"M94 80.8H40v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-302.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">a0</text></g><g class=\\"node\\"><path fill=\\"#fff\\" stroke=\\"#fff\\" d=\\"M94 152.8H40v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-230.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">a1</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M67 117.1v25.58\\"></path><path d=\\"m70.5 142.69-3.5 10-3.5-10h7z\\"></path></g><g class=\\"node\\"><path fill=\\"#fff\\" stroke=\\"#fff\\" d=\\"M94 224.8H40v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-158.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">a2</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M67 189.1v25.58\\"></path><path d=\\"m70.5 214.69-3.5 10-3.5-10h7z\\"></path></g><g class=\\"node\\"><path fill=\\"#d3d3d3\\" stroke=\\"#000\\" d=\\"M204 296.8h-54v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-86.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">b3</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"m80.27 188.92 77.25 99.73\\"></path><path d=\\"m160.33 286.56 3.35 10.05-8.89-5.77 5.54-4.28z\\"></path></g><g class=\\"node\\"><path fill=\\"#fff\\" stroke=\\"#fff\\" d=\\"M94 296.8H40v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-86.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">a3</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M67 261.1v25.58\\"></path><path d=\\"m70.5 286.69-3.5 10-3.5-10h7z\\"></path></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M51.39 296.57c-7.76-9.74-16.34-22.59-20.39-35.77-14.11-45.88-14.11-62.12 0-108 3.01-9.79 8.51-19.39 14.32-27.67\\"></path><path d=\\"m42.59 122.93 8.8-5.91-3.2 10.1-5.6-4.19z\\"></path></g><g class=\\"node\\"><path fill=\\"none\\" stroke=\\"#000\\" d=\\"M140.22 368.69h-36.44v36.43h36.44v-36.43zm-24.44 0-12 12m0 12.43 12 12m12.44 0 12-12m0-12.43-12-12\\"></path><text text-anchor=\\"middle\\" x=\\"118\\" y=\\"-13.91\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">end</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M80.6 333.13c6.52 8.32 14.47 18.44 21.65 27.59\\"></path><path d=\\"m105.14 358.74 3.42 10.03-8.93-5.71 5.51-4.32z\\"></path></g><g class=\\"node\\"><path fill=\\"#d3d3d3\\" stroke=\\"#000\\" d=\\"M204 80.8h-54v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-302.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">b0</text></g><g class=\\"node\\"><path fill=\\"#d3d3d3\\" stroke=\\"#000\\" d=\\"M204 152.8h-54v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-230.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">b1</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M177 117.1v25.58\\"></path><path d=\\"m180.5 142.69-3.5 10-3.5-10h7z\\"></path></g><g class=\\"node\\"><path fill=\\"#d3d3d3\\" stroke=\\"#000\\" d=\\"M204 224.8h-54v36h54v-36z\\"></path><text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-158.01\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">b2</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M177 189.1v25.58\\"></path><path d=\\"m180.5 214.69-3.5 10-3.5-10h7z\\"></path></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M150.09 260.92c-14.25 9.07-31.94 20.32-47.33 30.12\\"></path><path d=\\"m104.33 294.19-10.32 2.42 6.56-8.33 3.76 5.91z\\"></path></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M177 261.1v25.58\\"></path><path d=\\"m180.5 286.69-3.5 10-3.5-10h7z\\"></path></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M163.4 333.13c-6.52 8.32-14.47 18.44-21.65 27.59\\"></path><path d=\\"m144.37 363.06-8.93 5.71 3.42-10.03 5.51 4.32z\\"></path></g><g class=\\"node\\"><path fill=\\"none\\" stroke=\\"#000\\" d=\\"M122 4 82.76 22 122 40l39.24-18L122 4zM93.67 16.99V27m17.42 7.99h21.82M150.33 27V16.99M132.91 9h-21.82\\"></path><text text-anchor=\\"middle\\" x=\\"118\\" y=\\"-378.81\\" font-family=\\"Times,serif\\" font-size=\\"14\\" transform=\\"translate(4 405.01)\\">start</text></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M112.42 36.02c-7.4 10.07-17.87 24.31-26.89 36.58\\"></path><path d=\\"m88.31 74.72-8.74 5.99 3.1-10.13 5.64 4.14z\\"></path></g><g class=\\"edge\\" stroke=\\"#000\\"><path fill=\\"none\\" d=\\"M131.58 36.02c7.4 10.07 17.87 24.31 26.89 36.58\\"></path><path d=\\"m161.33 70.58 3.1 10.13-8.74-5.99 5.64-4.14z\\"></path></g></g></svg></figure>
<p>Some text after.</p>"
`;

exports[`remark-graphviz optimize option should optimize generated svg file with svgo 1`] = `
"<p>Some text before.</p>
<figure><img src=\\"/201e5d6a36c0d17cc401665c7988024f.svg\\" alt=\\"\\"></figure>
<p>Some text after.</p>"
`;

exports[`remark-graphviz outputPath option should generate image element and svg file 1`] = `
"<p>Some text before.</p>
<figure><img src=\\"/e756674193b6837ef58eeadc5ff5d7b4.svg\\" alt=\\"\\"></figure>
<p>Some text after.</p>"
`;

exports[`remark-graphviz should generate figure element with generated svg 1`] = `
"<p>Some text before.</p>
<figure><!--?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?-->

<!-- Generated by graphviz version 2.49.0 (20210829.0556)
 -->
<!-- Title: G Pages: 1 -->
<svg width=\\"224pt\\" height=\\"409pt\\" viewBox=\\"0.00 0.00 224.00 409.01\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
<g id=\\"graph0\\" class=\\"graph\\" transform=\\"scale(1 1) rotate(0) translate(4 405.01)\\">
<title>G</title>
<polygon fill=\\"white\\" stroke=\\"transparent\\" points=\\"-4,4 -4,-405.01 220,-405.01 220,4 -4,4\\"></polygon>
<g id=\\"clust1\\" class=\\"cluster\\">
<title>cluster_0</title>
<polygon fill=\\"lightgrey\\" stroke=\\"lightgrey\\" points=\\"8,-64.21 8,-357.01 98,-357.01 98,-64.21 8,-64.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"53\\" y=\\"-340.41\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">Hello</text>
</g>
<g id=\\"clust2\\" class=\\"cluster\\">
<title>cluster_1</title>
<polygon fill=\\"none\\" stroke=\\"blue\\" points=\\"138,-64.21 138,-357.01 208,-357.01 208,-64.21 138,-64.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-340.41\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">World</text>
</g>
<!-- a0 -->
<g id=\\"node1\\" class=\\"node\\">
<title>a0</title>
<polygon fill=\\"white\\" stroke=\\"white\\" points=\\"90,-324.21 36,-324.21 36,-288.21 90,-288.21 90,-324.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-302.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">a0</text>
</g>
<!-- a1 -->
<g id=\\"node2\\" class=\\"node\\">
<title>a1</title>
<polygon fill=\\"white\\" stroke=\\"white\\" points=\\"90,-252.21 36,-252.21 36,-216.21 90,-216.21 90,-252.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-230.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">a1</text>
</g>
<!-- a0&#45;&gt;a1 -->
<g id=\\"edge1\\" class=\\"edge\\">
<title>a0->a1</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M63,-287.91C63,-280.2 63,-270.93 63,-262.33\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"66.5,-262.32 63,-252.32 59.5,-262.32 66.5,-262.32\\"></polygon>
</g>
<!-- a2 -->
<g id=\\"node3\\" class=\\"node\\">
<title>a2</title>
<polygon fill=\\"white\\" stroke=\\"white\\" points=\\"90,-180.21 36,-180.21 36,-144.21 90,-144.21 90,-180.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-158.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">a2</text>
</g>
<!-- a1&#45;&gt;a2 -->
<g id=\\"edge2\\" class=\\"edge\\">
<title>a1->a2</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M63,-215.91C63,-208.2 63,-198.93 63,-190.33\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"66.5,-190.32 63,-180.32 59.5,-190.32 66.5,-190.32\\"></polygon>
</g>
<!-- b3 -->
<g id=\\"node8\\" class=\\"node\\">
<title>b3</title>
<polygon fill=\\"lightgrey\\" stroke=\\"black\\" points=\\"200,-108.21 146,-108.21 146,-72.21 200,-72.21 200,-108.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-86.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">b3</text>
</g>
<!-- a1&#45;&gt;b3 -->
<g id=\\"edge9\\" class=\\"edge\\">
<title>a1->b3</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M76.27,-216.09C95.51,-191.25 131.29,-145.06 153.52,-116.36\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"156.33,-118.45 159.68,-108.4 150.79,-114.17 156.33,-118.45\\"></polygon>
</g>
<!-- a3 -->
<g id=\\"node4\\" class=\\"node\\">
<title>a3</title>
<polygon fill=\\"white\\" stroke=\\"white\\" points=\\"90,-108.21 36,-108.21 36,-72.21 90,-72.21 90,-108.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"63\\" y=\\"-86.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">a3</text>
</g>
<!-- a2&#45;&gt;a3 -->
<g id=\\"edge3\\" class=\\"edge\\">
<title>a2->a3</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M63,-143.91C63,-136.2 63,-126.93 63,-118.33\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"66.5,-118.32 63,-108.32 59.5,-118.32 66.5,-118.32\\"></polygon>
</g>
<!-- a3&#45;&gt;a0 -->
<g id=\\"edge11\\" class=\\"edge\\">
<title>a3->a0</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M47.39,-108.44C39.63,-118.18 31.05,-131.03 27,-144.21 12.89,-190.09 12.89,-206.33 27,-252.21 30.01,-262 35.51,-271.6 41.32,-279.88\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"38.59,-282.08 47.39,-287.99 44.19,-277.89 38.59,-282.08\\"></polygon>
</g>
<!-- end -->
<g id=\\"node10\\" class=\\"node\\">
<title>end</title>
<polygon fill=\\"none\\" stroke=\\"black\\" points=\\"136.22,-36.32 99.78,-36.32 99.78,0.11 136.22,0.11 136.22,-36.32\\"></polygon>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"111.78,-36.32 99.78,-24.32 \\"></polyline>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"99.78,-11.89 111.78,0.11 \\"></polyline>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"124.22,0.11 136.22,-11.89 \\"></polyline>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"136.22,-24.32 124.22,-36.32 \\"></polyline>
<text text-anchor=\\"middle\\" x=\\"118\\" y=\\"-13.91\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">end</text>
</g>
<!-- a3&#45;&gt;end -->
<g id=\\"edge12\\" class=\\"edge\\">
<title>a3->end</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M76.6,-71.88C83.12,-63.56 91.07,-53.44 98.25,-44.29\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"101.14,-46.27 104.56,-36.24 95.63,-41.95 101.14,-46.27\\"></polygon>
</g>
<!-- b0 -->
<g id=\\"node5\\" class=\\"node\\">
<title>b0</title>
<polygon fill=\\"lightgrey\\" stroke=\\"black\\" points=\\"200,-324.21 146,-324.21 146,-288.21 200,-288.21 200,-324.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-302.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">b0</text>
</g>
<!-- b1 -->
<g id=\\"node6\\" class=\\"node\\">
<title>b1</title>
<polygon fill=\\"lightgrey\\" stroke=\\"black\\" points=\\"200,-252.21 146,-252.21 146,-216.21 200,-216.21 200,-252.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-230.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">b1</text>
</g>
<!-- b0&#45;&gt;b1 -->
<g id=\\"edge4\\" class=\\"edge\\">
<title>b0->b1</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M173,-287.91C173,-280.2 173,-270.93 173,-262.33\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"176.5,-262.32 173,-252.32 169.5,-262.32 176.5,-262.32\\"></polygon>
</g>
<!-- b2 -->
<g id=\\"node7\\" class=\\"node\\">
<title>b2</title>
<polygon fill=\\"lightgrey\\" stroke=\\"black\\" points=\\"200,-180.21 146,-180.21 146,-144.21 200,-144.21 200,-180.21\\"></polygon>
<text text-anchor=\\"middle\\" x=\\"173\\" y=\\"-158.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">b2</text>
</g>
<!-- b1&#45;&gt;b2 -->
<g id=\\"edge5\\" class=\\"edge\\">
<title>b1->b2</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M173,-215.91C173,-208.2 173,-198.93 173,-190.33\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"176.5,-190.32 173,-180.32 169.5,-190.32 176.5,-190.32\\"></polygon>
</g>
<!-- b2&#45;&gt;a3 -->
<g id=\\"edge10\\" class=\\"edge\\">
<title>b2->a3</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M146.09,-144.09C131.84,-135.02 114.15,-123.77 98.76,-113.97\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"100.33,-110.82 90.01,-108.4 96.57,-116.73 100.33,-110.82\\"></polygon>
</g>
<!-- b2&#45;&gt;b3 -->
<g id=\\"edge6\\" class=\\"edge\\">
<title>b2->b3</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M173,-143.91C173,-136.2 173,-126.93 173,-118.33\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"176.5,-118.32 173,-108.32 169.5,-118.32 176.5,-118.32\\"></polygon>
</g>
<!-- b3&#45;&gt;end -->
<g id=\\"edge13\\" class=\\"edge\\">
<title>b3->end</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M159.4,-71.88C152.88,-63.56 144.93,-53.44 137.75,-44.29\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"140.37,-41.95 131.44,-36.24 134.86,-46.27 140.37,-41.95\\"></polygon>
</g>
<!-- start -->
<g id=\\"node9\\" class=\\"node\\">
<title>start</title>
<polygon fill=\\"none\\" stroke=\\"black\\" points=\\"118,-401.01 78.76,-383.01 118,-365.01 157.24,-383.01 118,-401.01\\"></polygon>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"89.67,-388.02 89.67,-378.01 \\"></polyline>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"107.09,-370.02 128.91,-370.02 \\"></polyline>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"146.33,-378.01 146.33,-388.02 \\"></polyline>
<polyline fill=\\"none\\" stroke=\\"black\\" points=\\"128.91,-396.01 107.09,-396.01 \\"></polyline>
<text text-anchor=\\"middle\\" x=\\"118\\" y=\\"-378.81\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">start</text>
</g>
<!-- start&#45;&gt;a0 -->
<g id=\\"edge7\\" class=\\"edge\\">
<title>start->a0</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M108.42,-368.99C101.02,-358.92 90.55,-344.68 81.53,-332.41\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"84.31,-330.29 75.57,-324.3 78.67,-334.43 84.31,-330.29\\"></polygon>
</g>
<!-- start&#45;&gt;b0 -->
<g id=\\"edge8\\" class=\\"edge\\">
<title>start->b0</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M127.58,-368.99C134.98,-358.92 145.45,-344.68 154.47,-332.41\\"></path>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"157.33,-334.43 160.43,-324.3 151.69,-330.29 157.33,-334.43\\"></polygon>
</g>
</g>
</svg>
</figure>
<p>Some text after.</p>"
`;
